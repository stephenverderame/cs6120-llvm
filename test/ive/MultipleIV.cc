// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    int j = 0;
    for (int i = 5000; i >= m && j < 100; i -= m, ++j) {
        printf("%d\n", i);
    }
}
// NOLINTEND