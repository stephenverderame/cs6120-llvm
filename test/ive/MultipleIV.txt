5000
4980
4960
4940
4920
4900
4880
4860
4840
4820
4800
4780
4760
4740
4720
4700
4680
4660
4640
4620
4600
4580
4560
4540
4520
4500
4480
4460
4440
4420
4400
4380
4360
4340
4320
4300
4280
4260
4240
4220
4200
4180
4160
4140
4120
4100
4080
4060
4040
4020
4000
3980
3960
3940
3920
3900
3880
3860
3840
3820
3800
3780
3760
3740
3720
3700
3680
3660
3640
3620
3600
3580
3560
3540
3520
3500
3480
3460
3440
3420
3400
3380
3360
3340
3320
3300
3280
3260
3240
3220
3200
3180
3160
3140
3120
3100
3080
3060
3040
3020
