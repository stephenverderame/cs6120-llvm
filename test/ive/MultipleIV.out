Basic IV:   %3 = phi i32 [ 5000, %0 ], [ %6, %2 ]
Basic IV:   %4 = phi i32 [ 0, %0 ], [ %7, %2 ]
Basic IV:   %6 = add nsw i32 %3, -20
Basic IV:   %7 = add nuw nsw i32 %4, 1
